src/schema.rs

```diff
@@ -499,6 +509,112 @@ impl Serialize for RecordField {
     }
 }
 
+/// Parses a **valid** avro schema into the Parsing Canonical Form.
+/// https://avro.apache.org/docs/1.8.2/spec.html#Parsing+Canonical+Form+for+Schemas
+fn parsing_canonical_form(schema: &serde_json::Value) -> String {
+    match schema {
+        serde_json::Value::Number(n) => pcf_float(n),
+        serde_json::Value::Object(map) => pcf_map(map),
+        serde_json::Value::String(s) => pcf_string(s),
+        serde_json::Value::Array(v) => pcf_array(v),
+        _ => unreachable!(),
+    }
+}
+
+fn pcf_map(schema: &Map<String, serde_json::Value>) -> String {
+    use std::borrow::Cow;
```

Comment:
nit: do you mind moving this to the top of the file? Or is there any specific reason for having it here?

Reply:
Don't mind - done